src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -973,28 +974,19 @@ public String getCastScriptStatement(String name, String castType, List<KVValue>
      * approach will return type of result column as DOUBLE, although there is enough information to understand that
      * it might be safely treated as INTEGER.
      */
-    public static Schema.Type getScriptFunctionReturnType(Field field) {
+    public static Schema.Type getScriptFunctionReturnType(
+            int fieldIndex, Field field, ColumnTypeProvider scriptColumnType) {
```

Comment:
2nd argument should be `MethodField`. Then I think you should reorganize this method like this:

```
returnType
if (methodField is cast)
    returnType = ...
else
    returnType = ...

if (returnType == null)
    throw exception ...

return returnType
```

After this, could you add UT for this function because of its importance? Maybe 3 test cases for the `if`, `else` and `throw exception` in `SQLFunctionsTest`.